java/flight/src/main/java/org/apache/arrow/flight/FlightServer.java

```diff
@@ -90,20 +86,86 @@ public void close() throws InterruptedException {
     }
   }
 
-  public interface OutputFlight {
-    void sendData(int count);
+  public static Builder builder(BufferAllocator allocator, Location location, FlightProducer producer) {
+    return new Builder(allocator, location, producer);
+  }
 
-    void done();
+  public static final class Builder {
 
-    void fail(Throwable t);
-  }
+    private final BufferAllocator allocator;
+    private final Location location;
+    private final FlightProducer producer;
+    private ServerAuthHandler authHandler = ServerAuthHandler.NO_OP;
 
-  public interface FlightServerHandler {
+    private InputStream certChain;
+    private InputStream key;
 
-    public FlightInfo getFlightInfo(String descriptor) throws Exception;
+    Builder(BufferAllocator allocator, Location location, FlightProducer producer) {
+      this.allocator = allocator;
```

Comment:
these properties should be settable via builder methods (rather than just constructor)